test/Base_Tests/src/Data/Text/Regex_Spec.enso

```diff
@@ -525,7 +525,7 @@ add_specs suite_builder =
 
     suite_builder.group "caching" group_builder->
         group_builder.specify "Replacer cache drops old values" <|
-            pattern = Regex.compile('([a-c])')
+            pattern = Regex.compile '([a-c])'
```

Comment:
> As I believe that `('([a-c])')` is valid expression, right?

Sure.

> Then `Regex.compile ('([a-c])')` must be valid as well!

Yes, because that's application syntax: `<expression><space><expression>`.

> And `Regex.compile('([a-c])')` isn't that far from that, is it?

I think the missing space makes a big difference. Concatenating two expressions doesn't have to produce a valid expression.

> e.g. the change is intentional. I just don't know why?

If we allow this, the language has two application syntaxes: Spaced application, and a higher-precedence unspaced application that *looks like* the way functions are called in many other languages. I think the familiarity of the unspaced syntax may make it particularly confusing for new language users, because we *don't* actually support JS-style function calls; as soon as someone tries to add a second argument, like `f(x, y)`, they will get a syntax error. It seems to me an alternative syntax with some baggage. Is it worth that cost, for advanced users? I don't see the advantage. Usually when higher-precedence application is needed, it can be achieved with the same number of parentheses using the normal spaced-application syntax and a parenthesized group.

Reply:
> If we allow this, the language has two application syntaxes: Spaced application, and a higher-precedence unspaced application

No, this is not the interpretation I am proposing. I don't think we should have two syntaxes. There should be only one: application. These three, should in my opinion be identical:
- `f x`
- `f (x)`
- `f(x)`

They should all have the **same precedence**.

Reply:
I didn't realize the proposal was to keep the syntax but change the interpretation. I think that would be confusing in cases like `g f(x)`, which I don't think looks equivalent to `g f x`.